src/App.tsx

```diff
@@ -1,7 +1,39 @@
-import React from 'react';
+import { Component } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressKey: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    pressKey: 'Nothing was pressed yet',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      if (event.key) {
+        this.setState({ pressKey: `The last pressed key is [${event.key}]` });
+      }
+    });
+  }
```

Comment:
1. it's a good practice to move your handler to an own function
```
  const handleKeyPress = (event: KeyboardEvent) => {
    this.setState({ pressKey: `The last pressed key is [${event.key}]` });
  };
```

2. No need to use this condition 
`if (event.key)`

3. You can use this `handleKeyPress` handler as for `componentDidMount` and for `componentWillUnmount`